### ISSUE TITLE
Cannot read property hostEcm of undefined

### ISSUE THREAD

**Issue by jpotts** [Alfresco/alfresco-js-api#53]
If you go into an empty folder and do npm install --save alfresco-js-api then try running this JS file:
```
var AlfrescoApi = require('alfresco-js-api');

this.alfrescoJsApi = new AlfrescoApi();

this.alfrescoJsApi.login('admin', 'admin').then(function (data) {
    console.log('API called successfully Login ticket:' + data);
}, function (error) {
    console.error(error);
});
```
It results in this exception:
```
/Users/jpotts/Documents/code/alfresco-js/node_modules/alfresco-js-api/src/alfrescoApi.js:31
            hostEcm: config.hostEcm || 'http://127.0.0.1:8080',
                           ^

TypeError: Cannot read property 'hostEcm' of undefined
    at new AlfrescoApi (/Users/jpotts/Documents/code/alfresco-js/node_modules/alfresco-js-api/src/alfrescoApi.js:31:28)
    at Object.<anonymous> (/Users/jpotts/Documents/code/alfresco-js/test.js:3:22)
    at Module._compile (module.js:541:32)
    at Object.Module._extensions..js (module.js:550:10)
    at Module.load (module.js:458:32)
    at tryModuleLoad (module.js:417:12)
    at Function.Module._load (module.js:409:3)
    at Module.runMain (module.js:575:10)
    at run (node.js:348:7)
    at startup (node.js:140:9)
```

**Comment by magemello** [comment]
Hi Jeff,

Can you change your code to:

var AlfrescoApi = require('alfresco-js-api');

this.alfrescoJsApi = new AlfrescoApi({ provider:'ALL' });  //

this.alfrescoJsApi.login('admin', 'admin').then(function (data) {
    console.log('API called successfully Login ticket:' + data);
}, function (error) {
    console.error(error);
});

If you want to login only in the ECM or the BPM, instead of ALL you have to pass ECM or BPM.

**Comment by magemello** [comment]
The problem should be fixed in the latest release alfresco-js-api@0.3.2.

**Comment by jpotts** [comment]
I've confirmed that with 0.3.2 my original code runs as-is. Thanks for fixing.
